src/info.c

```diff
@@ -1273,6 +1273,43 @@ static void info_shiplogView( unsigned int wid, char *str )
       dialogue_msgRaw( _("Log message"), tmp);
 }
 
+/**
+ * @brief Asks the player for an entry to add to the log
+ *
+ * @param wid Window widget
+ * @param str Button widget name
+ */
+static void info_shiplogAdd( unsigned int wid, char *str )
+{
+   char *tmp;
+   char *logname;
+   int logid;
+   (void) str;
+
+   logname = toolkit_getList( wid, "lstLogs" );
+   if ( logname == NULL || !strcmp( "All", logname ) ){
+      tmp = dialogue_inputRaw( "Add a log entry", 0, 4096, "Add an entry to your diary:" );
+      if ( tmp != NULL && strlen ( tmp ) > 0 ){
+         if ( shiplog_getID( "Diary" ) == -1 )
+              shiplog_create( "Diary", "Your diary", "Diary", 0, 0 );
+         shiplog_append( "Diary", tmp );
+         free( tmp );
+      }
+   }else{
+      tmp = dialogue_input( "Add a log entry", 0, 4096, "Add an entry to the log titled '%s':", logname );
+      if ( tmp != NULL && strlen ( tmp ) > 0 ){
```

Comment:
Parentheses needed around the statements being &&'d, no space after strlen, insert space between closing parenthesis and curly brace.